pkg/util/installinfo/install_info.go

```diff
@@ -58,12 +80,60 @@ func GetFilePath(conf model.Reader) string {
 
 // Get returns information about how the Agent was installed.
 func Get(conf model.Reader) (*InstallInfo, error) {
+	if installInfo := getRuntimeOverride(); installInfo != nil {
+		return installInfo, nil
+	}
 	if installInfo, ok := getFromEnvVars(); ok {
 		return installInfo, nil
 	}
 	return getFromPath(GetFilePath(conf))
 }
 
+// setRuntimeInstallInfo sets the install info at runtime, overriding file and env var values
+func setRuntimeInstallInfo(info *InstallInfo) error {
+	if info == nil {
+		return fmt.Errorf("install info cannot be nil")
+	}
+
+	if info.Tool == "" || info.ToolVersion == "" || info.InstallerVersion == "" {
+		return fmt.Errorf("install info must have tool, tool_version, and installer_version set")
+	}
+
+	runtimeInfoMutex.Lock()
+	defer runtimeInfoMutex.Unlock()
+
+	runtimeInstallInfo = scrubFields(&InstallInfo{
+		Tool:             info.Tool,
+		ToolVersion:      info.ToolVersion,
+		InstallerVersion: info.InstallerVersion,
+	})
+
+	log.Infof("Runtime install info set: tool=%s, tool_version=%s, installer_version=%s",
+		runtimeInstallInfo.Tool, runtimeInstallInfo.ToolVersion, runtimeInstallInfo.InstallerVersion)
+
+	return nil
+}
+
+// getRuntimeInstallInfo returns the current runtime override if set
+func getRuntimeInstallInfo() *InstallInfo {
+	return getRuntimeOverride()
```

Comment:
Let's combine these two functions, doesn't seem worth it to have one just call the other.